src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlStoreClient.cs

```diff
@@ -130,7 +131,18 @@
             var resourceTypeId = reader.Read(VLatest.Resource.ResourceTypeId, 0);
             var resourceId = reader.Read(VLatest.Resource.ResourceId, 1);
             var resourceSurrogateId = reader.Read(VLatest.Resource.ResourceSurrogateId, 2);
-            var version = reader.Read(VLatest.Resource.Version, 3);
+            int version;
+            retryVersion:
+            try
+            {
+                version = versionIsLong ? (int)reader.GetInt64(3) : reader.GetInt32(3);
+            }
+            catch (InvalidCastException)
+            {
+                versionIsLong = !versionIsLong;
+                goto retryVersion;
+            }
```

Comment:
Please add static method to read version and share it across all 4 places.

Reply:
Done